lsdcGui.py

```diff
@@ -2280,13 +2280,16 @@ def createSampleTab(self):
         self.zoomRadioGroup.addButton(self.zoom1Radio)
         self.zoom2Radio = QtWidgets.QRadioButton("Mag2")
         self.zoom2Radio.toggled.connect(functools.partial(self.zoomLevelToggledCB,"Zoom2"))
-        self.zoomRadioGroup.addButton(self.zoom2Radio)
+        if (daq_utils.sampleCameraCount>=2):
+          self.zoomRadioGroup.addButton(self.zoom2Radio)
         self.zoom3Radio = QtWidgets.QRadioButton("Mag3")
         self.zoom3Radio.toggled.connect(functools.partial(self.zoomLevelToggledCB,"Zoom3"))
-        self.zoomRadioGroup.addButton(self.zoom3Radio)
+        if (daq_utils.sampleCameraCount>=3):
+          self.zoomRadioGroup.addButton(self.zoom3Radio)
         self.zoom4Radio = QtWidgets.QRadioButton("Mag4")
         self.zoom4Radio.toggled.connect(functools.partial(self.zoomLevelToggledCB,"Zoom4"))
-        self.zoomRadioGroup.addButton(self.zoom4Radio)
+        if (daq_utils.sampleCameraCount>=4):
```

Comment:
if you really wanted to clean up the logic, you could use nested ifs to add buttons, but this may be slightly more complicated to work through

if ... >= 2:
   if ...>= 3: